core/db/migrate/20190106184413_remove_code_from_spree_promotions.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+class RemoveCodeFromSpreePromotions < ActiveRecord::Migration[5.1]
+  class Spree::Promotion < ActiveRecord::Base
+    has_many :codes, class_name: "Spree::PromotionCode"
+  end
+
+  def up
+    promotions_with_code = Spree::Promotion.where.not(code: nil)
+
+    if promotions_with_code.any?
+      # You have some promotions with "code" field present! This is not good
+      # since we are going to remove that column.
+      #
+      self.class.promotions_with_code_handler.new(self, promotions_with_code).call
+    end
+
+    remove_index :spree_promotions, name: :index_spree_promotions_on_code
+    remove_column :spree_promotions, :code
+  end
+
+  def down
+    add_column :spree_promotions, :code, :string
+    add_index :spree_promotions, :code, name: :index_spree_promotions_on_code
+  end
+
+  def self.promotions_with_code_handler
+    # We propose different approaches, just pick the one that you prefer or
+    # write your custom one.
+    #
+    # The fist one (raising an exception) is the default but you can
+    # comment/uncomment the one then better fits you needs.
+    RaiseException
+    #
+    # You can also use one of the following:
+    #
+    # MoveToSpreePromotionCode
+    # DoNothing
+  end
+
+  class PromotionsWithCodeHandler
+    attr_reader :migration_context, :promotions
+
+    def initialize(migration_context, promotions)
+      @migration_context = migration_context
+      @promotions = promotions
+    end
+  end
+
+  class RaiseException < PromotionsWithCodeHandler
+    def call
+      # Please note that this will block the current migration and rollback all
+      # the previous ones run with the same "rails db:migrate" command.
+      #
+      raise StandardError, "You are trying to drop 'code' column from "\
+        "spree_promotions table but you have at least one record with that "\
+        "column filled. Please take care of that or you could lose data. See:" \
+        "\n" \
+        "https://github.com/solidusio/solidus/pull/3028"\
+        "\n"
+    end
+  end
+
+  class MoveToSpreePromotionCode < PromotionsWithCodeHandler
+    def call
+      # This is another possible approach, it will convert Spree::Promotion#code
+      # to a Spree::PromotionCode before removing the `code` field.
+      #
+      promotions.each do |promotion|
```

Comment:
I suppose the number of promotions might be quite high in some databases. In that case `find_each` should  work better than `each`